scripts/HPC_chatbot_preprocessor/chatbot_parser.py

```diff
@@ -0,0 +1,593 @@
+                line = line[:pos_first_mangle] + "-if-" + line[pos_first_mangle:pos_second_mangle] + "-if-" + line[pos_second_mangle:]
+                added_length += 8
+                if is_os == 2:
+                    is_os = 3
+                else:
+                    is_os = 2
+            else:
+                if is_os == 2:
+                    is_os = 1
+                else:
+                    is_os = 0
+        else:
+            if is_os == 2 or is_os == 3:
+                line = line[:pos_first_mangle] + "-if-" + line[pos_first_mangle:pos_second_mangle] + "-if-" + line[pos_second_mangle:]
+                added_length += 8
+
+        start_index += constr_match.end()
+        match = re.search(r'\{%(.*?)%}(.*)', match.group(2))
+    return line, is_os
+
+
+def mangle_ifs(directory, filename):
+    """
+    function that writes the if-mangled version of a file to a location where the jinja parser will use it
+
+    :param directory: the directory of the file to be if mangled
+    :param filename: the filename of the file to be mangled
+    :return:
+    """
+    # variable to keep track of latest if-statement scope
+    is_os = 0  # Can be 0, 1, 2 or 3 {0: not in an os-if; 1: in a non-os-if nested in an os-if; 2: in an os-if; 3: in an os-if nested in an os-if}
+
+    with open(os.path.join("if_mangled_files",  filename), 'w') as write_file:
+        with open(directory, 'r') as read_file:
+            for line in read_file:
+                new_line, is_os = mangle_os_ifs(line, is_os)
+                write_file.write(new_line)
+
+
+def check_if_statements(curr_line, active_OS_if_states):
+    """
+    function that checks for if-statements
+
+    :param curr_line: the line to be checked for if-statements to build the directory structure
+    :param active_OS_if_states: dictionary keeping track of the active OS states according to the if-statements
+    :return: the next action to be done with the line:
+                "done": An if-statement has been found at the start of the line, the active os list has been updated, processing of the current line is finished and a following line can be processed.
+                "check_extra_message": An if-statement has been found at the start of the line, the active os list has been updated, more text has been detected after the if-statement that also needs to be checked.
+                "write_text": No if-statement has been found, write the current line to a file (can also be part of the current line)
+                "write_text_and_check_extra_message": An if statement has been found not at the start of the line. Firstly, write the text up until the if-statement to a file, then check the rest of the line.
+    :return: the extra message to be checked, if any
+    :return: the text to be written to the file, if any
+    """
+    # check whether the first part of the line contains information wrt if-statements
+    match = re.search(r'^\{-if-%(.*?)%-if-}(.*)', curr_line)
```

Comment:
use constant for `-if-`, also in `mangle_os_ifs`:

```suggestion
    match = re.search(r'^\{' + MANGLED_IF + "%(.*?)%" + MANGLED_IF + "}(.*)', curr_line)
```